src/command_modules/azure-cli-vm/azure/cli/command_modules/vm/tests/latest/test_vm_commands.py

```diff
@@ -676,6 +676,23 @@ def test_vm_managed_disk(self, resource_group):
                      self.check('storageProfile.osDisk.caching', 'ReadWrite')
                  ])
 
+    @ResourceGroupPreparer(name_prefix='cli_test_large_disk')
+    def test_vm_larget_disk(self, resource_group):
+        self.kwargs.update({
+            'disk': 'd1',
+            'disk2': 'd2',
+            'snapshot': 's1'
+        })
+        self.cmd('disk create -g {rg} -n {disk} --size-gb 1 --hyper-v-generation V2 --for-upload', checks=[
+            self.check('hyperVgeneration', "V2")
+        ])
+        self.cmd('disk grant-access -g {rg} -n {disk} --access-level Write --duration-in-seconds 3600')
+        # To connfirm with compute team that snapshot indeed have --hyper-v-generation support
+        # self.cmd('disk create -g {rg} -n {disk2} --size-gb 1')
+        # self.cmd('snapshot create -g {rg} -n {snapshot} --source {disk2} --hyper-v-generation V2', checks=[
+        #     self.check('hyperVgeneration', "V2")
+        # ])
+
```

Comment:
Will an end-to-end test where we upload some data to a large disk be helpful or overkill?